src/main/java/edu/wpi/first/deployutils/deploy/artifact/NativeLibraryArtifact.java

```diff
@@ -0,0 +1,62 @@
+package edu.wpi.first.deployutils.deploy.artifact;
+
+import java.io.File;
+
+import javax.inject.Inject;
+
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.util.PatternFilterable;
+import org.gradle.api.tasks.util.PatternSet;
+import org.gradle.nativeplatform.NativeLibraryBinarySpec;
+
+import edu.wpi.first.deployutils.deploy.cache.CacheMethod;
+import edu.wpi.first.deployutils.deploy.context.DeployContext;
+import edu.wpi.first.deployutils.deploy.target.RemoteTarget;
+
+public class NativeLibraryArtifact extends AbstractArtifact implements CacheableArtifact {
+
+    private final Property<CacheMethod> cacheMethod;
+
+    @Inject
+    public NativeLibraryArtifact(String name, RemoteTarget target) {
+        super(name, target);
+        filename = target.getProject().getObjects().property(String.class);
+        binarySpec = target.getProject().getObjects().property(NativeLibraryBinarySpec.class);
```

Comment:
This should be limited to SharedLibraryBinarySpec. It doesn't make sense to deploy a static library.

That then gets you access to SharedLibraryBinarySpec.getTasks(), which has a getLink() method that gets you access to the link task, to use as the dependency.

https://docs.gradle.org/current/javadoc/org/gradle/nativeplatform/SharedLibraryBinarySpec.TasksCollection.html

Reply:
This also gets you access to getSharedLibraryFile(), which is the file you'd actually deploy (What comes from the install task for executables).

https://docs.gradle.org/current/javadoc/org/gradle/nativeplatform/SharedLibraryBinarySpec.html